### PR TITLE
Bschnurr/fix subpackages pytest

### LINKED ISSUE #5525: Pytest execution failure - tests in subpackages
image

Output:
```
[8/1/2019 6:06:04.914 PM Informational] Store opened in 0.004 sec.
[8/1/2019 6:06:08.829 PM Informational] ---------- Discovery started ----------
[8/1/2019 6:06:10.566 PM Informational] Starting pytest discovery for the PythonApplication84 project. This can take up to 60 seconds.
[8/1/2019 6:06:10.581 PM Informational] cd C:\Users\huvalo\source\repos\PythonApplication84\
[8/1/2019 6:06:10.605 PM Informational] set PYTHONPATH=C:\Users\huvalo\source\repos\PythonApplication84\
[8/1/2019 6:06:10.637 PM Informational] C:\Program Files\Python36\python.exe C:\USERS\HUVALO\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\16.0_D1270DCBEXP\EXTENSIONS\MICROSOFT CORPORATION\PYTHON\16.0.0\PythonFiles\testing_tools\run_adapter.py discover pytest -- --cache-clear c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test1.py c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test2.py c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\__init__.py c:\users\huvalo\source\repos\pythonapplication84\package1\__init__.py
[8/1/2019 6:06:11.661 PM Error] Failed to parse: Unexpected character encountered while parsing value: =. Path '', line 0, position 0.
[8/1/2019 6:06:11.696 PM Error] ============================= test session starts =============================
platform win32 -- Python 3.6.8, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: C:\Users\huvalo\source\repos\PythonApplication84
collected 0 items / 3 errors
=================================== ERRORS ====================================
_________________ ERROR collecting package1/packagea/test1.py _________________
ImportError while importing test module 'c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test1.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
ModuleNotFoundError: No module named 'package1.packagea'
_________________ ERROR collecting package1/packagea/test2.py _________________
ImportError while importing test module 'c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test2.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
ModuleNotFoundError: No module named 'package1.packagea'
_________________ ERROR collecting package1/packagea/test1.py _________________
ImportError while importing test module 'c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test1.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
ModuleNotFoundError: No module named 'package1.packagea'
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 3 error in 0.20 seconds ===========================
Traceback (most recent call last):
  File "C:\USERS\HUVALO\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\16.0_D1270DCBEXP\EXTENSIONS\MICROSOFT CORPORATION\PYTHON\16.0.0\PythonFiles\testing_tools\run_adapter.py", line 16, in <module>
    main(tool, cmd, subargs, toolargs)
  File "C:\USERS\HUVALO\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\16.0_D1270DCBEXP\EXTENSIONS\MICROSOFT CORPORATION\PYTHON\16.0.0\PythonFiles\testing_tools\adapter\__main__.py", line 122, in main
    parents, result = run(toolargs, **subargs)
  File "C:\USERS\HUVALO\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\16.0_D1270DCBEXP\EXTENSIONS\MICROSOFT CORPORATION\PYTHON\16.0.0\PythonFiles\testing_tools\adapter\__main__.py", line 35, in pytest_discover
    return pytest.discover(pytestargs, _ignored)
  File "C:\USERS\HUVALO\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\16.0_D1270DCBEXP\EXTENSIONS\MICROSOFT CORPORATION\PYTHON\16.0.0\PythonFiles\testing_tools\adapter\pytest\_discovery.py", line 34, in discover
    raise Exception('pytest discovery failed (exit code {})'.format(ec))
Exception: pytest discovery failed (exit code 2)

[8/1/2019 6:06:11.725 PM Warning] No test is available in c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test1.py c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\test2.py c:\users\huvalo\source\repos\pythonapplication84\package1\packagea\__init__.py c:\users\huvalo\source\repos\pythonapplication84\package1\__init__.py. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
[8/1/2019 6:06:11.740 PM Informational] ========== Discovery finished: 0 tests found (0:00:02.8603913) ==========

```

From command line pytest:
```
C:\Users\huvalo\source\repos\PythonApplication84>pytest package1\packageA\test1.py package1\packageA\test2.py
================================================= test session starts =================================================
platform win32 -- Python 3.6.8, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: C:\Users\huvalo\source\repos\PythonApplication84
collected 3 items

package1\packageA\test1.py .                                                                                     [ 33%]
package1\packageA\test2.py F.                                                                                    [100%]

====================================================== FAILURES =======================================================
__________________________________________________ Test_test2.test_A __________________________________________________

self = <package1.packageA.test2.Test_test2 testMethod=test_A>

    def test_A(self):
>       self.fail("todo implement this test")
E       AssertionError: todo implement this test

package1\packageA\test2.py:5: AssertionError
========================================= 1 failed, 2 passed in 0.09 seconds ==========================================

```

### PR DESCRIPTION
**Pull request by bschnurr**
- Fixing how we parse pytestId from classname so that it handles filenames with relative paths.
- Added new unit tests

Pytest execution failure - tests in subpackages Fix #5525

### CODE REVIEW

File: Python/Product/TestAdapter.Executor/Pytest/TestResultParser.cs [hunk 1]
```diff
@@ -55,13 +55,18 @@ public class TestResultParser {
...
+        ///   pytestID .\package1\packageA\test1.py::Test_test1::test_A
+        /// 
         /// </summary>
         /// <param name="filename"></param>
         /// <param name="classname"></param>
         /// <param name="funcname"></param>
         /// <returns></returns>
         internal static string CreatePytestId(string filename, string classname, string funcname) {
-            var classIndex = classname.IndexOf(".");
+            var classIndex = classname.LastIndexOf(".");
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Pytest/TestResultParser.cs]: I could see this failing if we can have classes inside classes.. but i dont think pytest supports that..

File: Python/Tests/TestAdapterTests/TestResultParserTests.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.PythonTools.TestAdapter.Pytest;
+
+namespace TestAdapterTests {
+    /// <summary>
+    /// Summary description for TestResultParserTests
+    /// </summary>
+    [TestClass]
+    public class TestResultParserTests {
+        public TestResultParserTests() {
```
Comment (huguesv) [inline Python/Tests/TestAdapterTests/TestResultParserTests.cs]: You should get rid of all these useless comments and unused code.

Comment (bschnurr) [reply Python/Tests/TestAdapterTests/TestResultParserTests.cs]: will do

Comment (huguesv) [reply Python/Tests/TestAdapterTests/TestResultParserTests.cs]: oh you're missing the copyright header too

Comment (bschnurr) [reply Python/Tests/TestAdapterTests/TestResultParserTests.cs]: good catch

File: Python/Tests/TestAdapterTests/TestResultParserTests.cs [hunk 3]
```diff
@@ -24,18 +24,36 @@ public class TestResultParserTests {
 
         [TestMethod]
         public void CreatePytestId_FuncInsideClass() {
-            Assert.AreEqual(".\\test2.py::Test_test2::test_A", TestResultParser.CreatePytestId("test2.py", "test2.Test_test2", "test_A"));
+            Assert.AreEqual(
```
Comment (huguesv) [inline Python/Tests/TestAdapterTests/TestResultParserTests.cs]: style: put the `);` on the next line, indented -1
```
hello(
    1,
    2,
    3
);
```

this matches the style used with curly braces:
```
void hello() {
    var a = 1;
}
```

same comment applies below

Comment (bschnurr) [reply Python/Tests/TestAdapterTests/TestResultParserTests.cs]: ok

**Pull request merged**